packages/beacon-node/src/chain/blocks/importBlock.ts

```diff
@@ -93,19 +88,12 @@ export async function importBlock(chain: ImportBlockModules, fullyVerifiedBlock:
   //
   // current justified checkpoint should be prev epoch or current epoch if it's just updated
   // it should always have epochBalances there bc it's a checkpoint state, ie got through processEpoch
-  const justifiedCheckpoint = postState.currentJustifiedCheckpoint;
-
-  const onBlockPrecachedData: OnBlockPrecachedData = {
-    executionStatus,
-    blockDelaySec: (Math.floor(Date.now() / 1000) - postState.genesisTime) % chain.config.SECONDS_PER_SLOT,
-  };
-  if (justifiedCheckpoint.epoch > chain.forkChoice.getJustifiedCheckpoint().epoch) {
-    const state = getStateForJustifiedBalances(chain, postState, block);
-    onBlockPrecachedData.justifiedBalances = getEffectiveBalanceIncrementsZeroInactive(state);
-  }
-
   const prevFinalizedEpoch = chain.forkChoice.getFinalizedCheckpoint().epoch;
-  chain.forkChoice.onBlock(block.message, postState, onBlockPrecachedData);
+  const blockDelaySec = (Math.floor(Date.now() / 1000) - postState.genesisTime) % chain.config.SECONDS_PER_SLOT;
+  const blockRoot = toHexString(chain.config.getForkTypes(block.message.slot).BeaconBlock.hashTreeRoot(block.message));
+  // Should compute checkpoint balances before forkchoice.onBlock
+  chain.checkpointBalancesCache.processState(blockRoot, postState);
```

Comment:
Calling fc_store.onVerifiedBlock() in a similar pattern as lighthouse sounds better than here.